src/realm/query_engine.hpp

```diff
@@ -308,13 +310,21 @@ class ParentNode {
         }
     }
 
+    void do_verify_column(const ColumnBase* col, size_t col_ndx = npos) const
+    {
+        if (col_ndx == npos)
+            col_ndx = m_condition_column_idx;
+        if (m_table && (col_ndx != npos)) {
```

Comment:
Extra parens scream out that there's something interesting going on with precedence. There's nothing special here, so they're misleading.